### ISSUE TITLE
Can I train fastconformer cache aware streaming architecture with TDT decoder?

### ISSUE THREAD

**Issue by hassan-webm** [NVIDIA-NeMo/NeMo#14291]
Here would it work if I define loss and decoding to TDT?
https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/fastconformer/cache_aware_streaming/fastconformer_transducer_bpe_streaming.yaml

similar to this:
https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/conformer/tdt/conformer_tdt_bpe.yaml

Will it work? Any guide or leads on this?

**Comment by JRD971000** [comment]
Hi @pzelasko @stevehuang52, I assume you would be the right person to look into this?

**Comment by stevehuang52** [comment]
I haven't tried, but the code should work, since the cache-aware part is only related to the encoder, while TDT/RNNT/CTC shouldn't matter.

**Comment by hassan-webm** [comment]
@stevehuang52 and what training script can I use?

**Comment by zhenyih** [comment]
ai generated, please verify

Yes, you can train a FastConformer cache-aware streaming architecture with a TDT decoder instead of the default Transducer decoder. 

To do this, you'll need to modify the fastconformer_transducer_bpe_streaming.yaml file by:

1. Adding TDT-specific model defaults:
   ```
   model_defaults:
     tdt_durations: [0, 1, 2, 3, 4]
     num_tdt_durations: 5
   ```

2. Changing the loss section to use TDT loss:
   ```
   loss:
     loss_name: "tdt"
     tdt_kwargs:
       durations: ${model.model_defaults.tdt_durations}
       sigma: 0.05
       omega: 0.1
       fastemit_lambda: 0.001
   ```

3. Adding to the joint network configuration:
   ```
   jointnet:
     num_extra_outputs: ${model.model_defaults.num_tdt_durations}
   ```

4. Updating the decoding section:
   ```
   decoding:
     strategy: "greedy"
     model_type: "tdt"
     durations: ${model.model_defaults.tdt_durations}
   ```

The cache-aware streaming functionality affects the encoder, while TDT modifies the decoder and loss function, so they're compatible components that can work together.

**Comment by nithinraok** [comment]
Hi you can use this training script https://github.com/NVIDIA-NeMo/NeMo/blob/main/examples/asr/asr_transducer/speech_to_text_rnnt_bpe.py with configs from https://github.com/NVIDIA-NeMo/NeMo/tree/main/examples/asr/conf/fastconformer/cache_aware_streaming 

You can also find relevant training scripts based on type of decoder from here https://github.com/NVIDIA-NeMo/NeMo/tree/main/examples/asr

**Comment by KunalDhawan** [comment]
Hi @hassan-webm,
We haven’t heard back from you for a while, so I’ll go ahead and close this issue for now.
If you’re still experiencing the problem or have any updates to share, please feel free to reopen the issue and we’ll be happy to take another look.
